src/main/java/uk/gov/hmcts/reform/notificationservice/service/NotificationService.java

```diff
@@ -84,10 +84,12 @@ private void processNotifications(Notification notification) {
 
             notificationRepository.markAsFailure(notification.id);
         } catch (FeignException exception) {
+            var status = HttpStatus.resolve(exception.status());
+
             log.error(
                 "Received {} from client. Postponing notification for later. "
                     + "Service: {}, Zip file: {}, ID: {}, Client response: {}",
-                HttpStatus.valueOf(exception.status()).getReasonPhrase(),
+                (status == null ? HttpStatus.INTERNAL_SERVER_ERROR : status).getReasonPhrase(),
```

Comment:
I think of the status is null we should print sth custom instead of  internal server error  so we spot the difference.

Reply:
i don't mind amending. on the other hand, it's serious failure anyway and we will address it and/or see it because log includes exception as well. status is not as important as client response at the end of message ;).

another idea: if we cannot translate status into spring one so we can get human readable phrase, shall i just stick in actual number which comes back from `FeignException::status`?

Reply:
Agree. There is a 418 HttpStatus value 'I am a teapot', sounds a bit funny but maybe a good fit for this purpose - will not clash with any meaningful value

Reply:
just write log sth meaningful.  do we need to find use http status?

Reply:
it makes sense to introduce why it is postponed. for me `-1` will not make sense but if we ok to put numbers instead of human readable messages - i'm fine

Reply:
if null  log  "Unknown error -1" ?

Reply:
I think actual numbers would be sufficient. And if status is null, it would be ok to see null in the message. That would give us the most accurate picture of what happened.